runelite-client/src/main/java/net/runelite/client/plugins/opponentinfo/OpponentInfoPlugin.java

```diff
@@ -96,7 +113,47 @@ else if (worldType.contains(WorldType.SEASONAL_DEADMAN))
 		}
 	}
 
-	public static Map<String, Integer> loadNpcHealth()
+	@Subscribe
+	public void onInteractingChanged(InteractingChanged event)
+	{
+		if (event.getSource() != client.getLocalPlayer())
+		{
+			return;
+		}
+
+		Actor opponent = event.getTarget();
+		if (opponent == null)
+		{
+			return;
+		}
+
+		lastOpponent = opponent;
+		lastTime = Instant.now();
```

Comment:
So this starts timeout when player starts fighting opponent? It should start when it stops probably as that is the point of timer